### ISSUE TITLE
Feature Request : Allow tuning or disabling of trusted machine settings

### ISSUE THREAD

**Issue by darryl-h** [Ylianst/MeshCentral#1262]
# Feature Request:
In https://github.com/Ylianst/MeshCentral/issues/691 the ability to remember a browser was added, however, we would like to remove this ability because of the security implications. 

In the following examples, by "tuning", I mean allowing the duration (in days) that the setting would be valid for could be tuned to the organizations needs

# Ask:
**1) Add settings in the "settings" section of the config where the feature could be disabled/tuned**

For example to disable for all sites:
```
 "settings": {
    "Port": 443,
    "Allow2FactorTrustedBrowser": false
  }
```
To change from 30 days to 7 days for all sites:
```
 "settings": {
    "Port": 443,
    "Allow2FactorTrustedBrowser": true,
    "2FactorTrustedDurationDays": 7
 }   
```

**2) Add settings in the "domains" that would override the global setting 
allowing disabling/tuning**

In this example, it would be off by default, but allowed by one sub domain, "MyDomain" and have the default setting (30 days)

```
 "settings": {
    "Port": 443,
    "Allow2FactorTrustedBrowser": false
 },   
 "MyDomain": {
      "Title": "My Test Domain",
      "UserQuota": 1048576,
      "MeshQuota": 248576,
      "Allow2FactorTrustedBrowser": true
    }
```

In this example, it would be off by default, all sub domains would inherit the TrustedDuration of 14 days, and one sub domain "MyDomain", would have it enabled  for 14 days.
```
 "settings": {
    "Port": 443,
    "Allow2FactorTrustedBrowser": false,
    "2FactorTrustedDurationDays": 14
 },   
 "MyDomain": {
      "Title": "My Test Domain",
      "UserQuota": 1048576,
      "MeshQuota": 248576,
      "Allow2FactorTrustedBrowser": true
    }
```
In this example, it would be on by default, and all sub domains would have the default 14 days, but one sub domain "MyDomain", would only allow it for 7 days.

```
 "settings": {
    "Port": 443,
    "Allow2FactorTrustedBrowser": true,
    "2FactorTrustedDurationDays": 14
 },   
 "MyDomain": {
      "Title": "My Test Domain",
      "UserQuota": 1048576,
      "MeshQuota": 248576,
      "Allow2FactorTrustedBrowser": true,
      "2FactorTrustedDurationDays": 7
    }
```

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.18 with this feature included but a little different. You can add the following line in the domain section:

```
"twoFactorCookieDurationDays": 14,
```

The default is 30, but you can set it to zero to disable this feature. Also, I had to use "two" instead of "2" given JavaScript limitations that don't allow a variable to start with a number.

For settings being the default for all domains, I really must try to keep settings for just server wide settings and domain for domain stuff. However, if I added a section called "DomainDefaults" that would include settings you want for all domains, I think I can add that. This way, you you can set the domain default separately from server settings.

This said, feedback welcome.

**Comment by darryl-h** [comment]
**Thanks so much, this is a great implementation!!**

Also, the idea of `DomainDefaults` would be awesome as well! This would be perfect not only for this, but also for `UserSessionIdleTimeout`!

If a setting is set in both the `DomainDefault` and in the `Domain`, I assume/am hoping that the setting in `Domain` would take precedence?

**Comment by Ylianst** [comment]
Checked in the improvement in GitHub, will be published in MeshCentral v0.5.1\9 where you can do this:

```
{
  "settings": {
    "Cert": "meshcentral.com",
  },
  "domaindefaults": {
    "title": "MyDefaultTitle"
  },
  "domains": {
    "": {
      "title": "BetterTitle",
    },
    "customer1": {
    }
  }
}
```

Domain defaults are copied to all domains unless the setting is already specified in the domain.

**Comment by darryl-h** [comment]
**Thanks so much for all your hard work!!** This update will allow the following (for us) instead of having it in each domain, which is awesome and completely unexpected. :) 

```
"domaindefaults": {
  "twoFactorCookieDurationDays": 0,
  "UserSessionIdleTimeout": 30,
  "UserQuota": 1048576,
  "MeshQuota": 248576,
  "NewAccountEmailDomains": [
    "OurCompany.com"
  ],
  "PasswordRequirements": {
    "min": 8,
    "max": 128,
    "upper": 1,
    "lower": 1,
    "numeric": 1,
    "nonalpha": 1,
    "reset": 90,
    "force2factor": true
  }
}
```

The secondary benefit is this makes maintenance **so much** easier, we can now tune `PasswordRequirements` in one place rather than N. 

This also makes adding domains a simple `/usr/bin/node /opt/meshcentral/node_modules/meshcentral/meshctrl.js Config --adddomain NewDomainName` and everything else is inherited. **So good...**

Let me know if you want to file a separate feature request for the `DomainDefaults`

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.19 with support for "domaindefaults". It works exactly as your example above. Let me know if it works, no need to a separate issue (but thanks for offering).

**Comment by darryl-h** [comment]
**outstanding job!** This has simplified our configuration 10x. Can't thank you enough!!

## DomainDefaults
Tested MeshCentral v0.5.19, seems to work **exactly** as you describe, setting the `domaindefaults` works as expected and settings are inherited, additionally, if the subdomain has the same setting, the setting is overridden as expected.

## twoFactorCookieDurationDays 0
Tested MeshCentral v0.5.19, works as expected, no `Remember this device for N days.` is observed during the token entering process.

## twoFactorCookieDurationDays N
Tested MeshCentral v0.5.19, set this to 7 days, observed `Remember this device for 7 days.` under the login token.
